File: solutions/987.reveal-cards-in-increasing-order/reveal-cards-in-increasing-order.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        int n = deck.length;
        Arrays.sort(deck); 
        int[] res = new int[n];
        Queue<Integer> queue = new LinkedList<>();
        for(int i = 0; i < n; i++) queue.add(i);
        for(int num : deck) {
            res[queue.poll()] = num;
            queue.add(queue.poll());
        }
        return res;
    }
}

File: solutions/987.reveal-cards-in-increasing-order/reveal-cards-in-increasing-order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        //Just do the Counter Operation
		int n = deck.size();
		vector<int> res(n);
		sort(deck.begin(), deck.end());
		res[n-1] = deck.back();
		for(int i = n - 2; i >= 0; --i){
			res[i] = deck[i];
			rotate(res.begin() + i + 1, res.end() - 1, res.end());
		}
		return res;
    }
};